src/edu/ucsb/nceas/metacat/dataone/SystemMetadataFactory.java

```diff
@@ -680,6 +687,33 @@ private static void reindexDataFile(Identifier id, SystemMetadata sysmeta) {
             //e.printStackTrace();
         }
 	}
+	
+	/**
+	 * Check if the the solr doc for the given id exists.
+	 * @param id
+	 * @param session
+	 * @param request
+	 * @return
+	 * @throws InvalidToken
+	 * @throws ServiceFailure
+	 * @throws NotAuthorized
+	 * @throws InvalidRequest
+	 * @throws NotImplemented
+	 * @throws NotFound
+	 * @throws IOException
+	 */
+	private static boolean solrDocExists(String id, Session session, HttpServletRequest request) throws InvalidToken, 
+	                                            ServiceFailure, NotAuthorized, InvalidRequest, NotImplemented, NotFound, IOException {
+	    boolean exists = false;
+	    String query = "q=id:" +id;
```

Comment:
Does this work well with special characters? Normally you'd do `q="${ID}"` to make sure something odd doesn't happen when an identifier, for example, contains an `&`.

Reply:
I will change it.